packager/signer/signer.go

```diff
@@ -466,10 +466,10 @@ type SXGParams struct {
 // consumed body, not a stream reader. Signer loads the whole payload in memory
 // in order to be able to sign it, because it's required by signer's
 // serveSignedExchange method, specifically by its underlying calls to
-// transformer.Process and to signedexchange.NewExchange. The former validates
-// the AMP document, the latter signs it. Note: the latter requires the whole
-// payload in memory, because MICE requires the sender to process its payload in
-// reverse order
+// transformer.Process and to signedexchange.NewExchange. The former performs
+// AMP HTML transforms, which depend on a non-streaming HTML parser. The latter
```

Comment:
Great, thanks for adding the reasoning for "former" too, now it's symmetrical :)